### PR TITLE
ICDS Dashboard: Basic port of AWC reports to mobile

### PR DESCRIPTION
**Pull request by czue**
image

Review by commit. I'll do a self review now as well.

### CODE REVIEW

File: custom/icds_reports/templates/icds_reports/dashboard.html [hunk 1]
```diff
@@ -95,7 +95,6 @@
 
         <script src="{% static 'js/directives/awc-opened-yesterday/awc-opened-yesterday.directive.js' %}"></script>
         <script src="{% static 'js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js' %}"></script>
-        <script src="{% static 'js/directives/awc-reports/awc-reports.directive.js' %}"></script>
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/dashboard.html]: this commit also should have updated `mocha.html` but I missed that until the end.

File: custom/icds_reports/static/js/icds_app_mobile.js [hunk 2]
```diff
@@ -47,8 +47,15 @@ MainMobileController.$inject = [
...
 ];
 
-// ui.bootstrap not truly needed - but location directive depends on it to compile
-window.angular.module('icdsApp', ['ngRoute', 'cgBusy', 'ui.bootstrap', 'datamaps', 'nvd3'])
+window.angular.module('icdsApp', [
+    'ngRoute', 'cgBusy', 'datamaps', 'nvd3',
+    // these libraries aren't truly needed but do to code sharing with the web dashboard,
+    // some directives depend on them to compile.
+    // in the future, ideally those directives would be refactored such that the web dependencies
+    // don't leak into mobile
```
Comment (czue) [inline custom/icds_reports/static/js/icds_app_mobile.js]: this is annoying but I'm not going to worry about it for now.

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/awc-reports.html [hunk 3]
```diff
@@ -0,0 +1,80 @@
+{% load staticfiles %}
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/awc-reports.html]: this is basically a direct import of this file https://github.com/dimagi/mobile-dashboard-prototype/pull/17/files#diff-f6e8c66fb33aef14103a9c901c4f54da

it completely changed through the course of this PR and ended up around only 20 functional lines so probably isn't worth reviewing here.

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/map-chart.html [hunk 4]
```diff
@@ -1,24 +1,10 @@
 {% load staticfiles %}
 <div class="app-bg min-vh-100">
   <div ng-show="!$ctrl.filtersOpen">
+    <div class="top-section p-3">
+      {% include 'icds_reports/icds_app/mobile/partials/mobile-heading.html' %}
+    </div>
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/map-chart.html]: this is more than just a refactor - it changes how navigation works on map / chart pages. addresses this card: https://trello.com/c/YE5QGErC/239-decide-on-final-navigation-plan

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js [hunk 5]
```diff
@@ -1697,8 +1697,13 @@ var weight_for_height = {
 
 var url = hqImport('hqwebapp/js/initial_page_data').reverse;
 
-function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOptionsBuilder, DTColumnBuilder, $compile, storageService, userLocationId, haveAccessToAllLocations, haveAccessToFeatures, isAlertActive) {
+function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOptionsBuilder, DTColumnBuilder,
+        $compile, storageService, dateHelperService, baseControllersService, userLocationId,
+        haveAccessToAllLocations, haveAccessToFeatures, isAlertActive, isMobile) {
```
[the remark this answers is not in the document]
Comment (czue) [reply custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: hmm, i guess I was a little overeager in expecting to need this

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js [hunk 6]
```diff
@@ -2600,18 +2600,6 @@ function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOp
         return i;
     };
 
-    vm.moveToLocation = function (loc, index) {
```
Comment (czue) [inline custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: 2 down, 3 to go of 100% copy/pasted declarations of this method.

Comment (orangejenny) [reply custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: Keep fighting the good fight

**Review by orangejenny — APPROVED** [review]

**Comment by czue** [comment]
build before the last commit (which was just lint) passed so killing the build and merging

**Pull request merged**